wicket-core/src/main/java/org/apache/wicket/markup/html/image/ExternalImage.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.markup.html.image;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.wicket.markup.ComponentTag;
+import org.apache.wicket.markup.html.WebComponent;
+import org.apache.wicket.markup.html.image.Image.Cors;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.Model;
+
+/**
+ * A component to display external images. The src / srcset information are hold in models
+ * 
+ * @author Tobias Soloschenko
+ *
+ */
+public class ExternalImage extends WebComponent
+{
+
+	private static final long serialVersionUID = 1L;
+
+	/** The x values to be used within the srcset */
+	private List<String> xValues = null;
+
+	/** The sizes of the responsive images */
+	private List<String> sizes = null;
+
+	/**
+	 * Cross origin settings
+	 */
+	private Cors crossOrigin = null;
+
+	private IModel<List<Serializable>> srcSetModels;
+
+	/**
+	 * Creates an external image
+	 * 
+	 * @param id
+	 *            the component id
+	 * @param src
+	 *            the source URL
+	 */
+	public ExternalImage(String id, Serializable src)
+	{
+		this(id, src, (List<Serializable>)null);
```

Comment:
If I am not wrong, it usually considered as a best practice to supply an empty list instead of a null list. I guess you can use `Collections.emptyList()` here.